### ISSUE TITLE
Attributes - Need the ability to select which account ID(s) to include in the TCert attribute

### ISSUE THREAD

**Issue by frankyclu** [hyperledger-archives/fabric#1844]
## Description

The current tcert-attribute based design allows user to select which attribute to include in the TCert. However, the attributes are pre-defined. So for any given attribute  (e.g. account IDs), user can only choose to either include or not include in his TCert.

This strategy works if attribute values are static. However, account IDs are expected to be dynamic overtime in many use cases (for obfuscation purposes .... e.g. bitcoin addresses are dynamically created from pub keys generated/managed by wallets).

With the current design, a user's entire account information will be exposed to whoever with the ability to decrypt the attributes; (there are a few ugly work around options such as to use multiple attributes to store Account IDs, and pass the name of the attributes along with certificate)
## Describe How to Reproduce

This chaincode example use attribute to store an entity's Account ID to be used in an asset depository ledger:
https://github.com/hyperledger/fabric/tree/master/examples/chaincode/go/asset_management_with_roles

**Comment by smithbk** [comment]
@frankyclu Requiring that attributes be listed in the membersrvc.yaml file is temporary.  One option being discussed is to provide an RPC call similar to ECAA.RegisterUser (see fabric/membersrvc.proto file) which allows adding/removing attributes to a particular user.  For some attributes such as "bank1.bankAccountNumber", it may be permissible to allow a single user from bank1 to add such an attribute.  A user could be limited in the attribute names that he/she could create (e.g. must start with "bank1.").  One question though is, "Are there attributes which must be attested to by multiple banks/entities on the chain?".  I can only think of use cases in which a single attribute is associated with a single entity on a blockchain.  However, it there is a need for approval or attestation from multiple entities on a block chain before granting a particular user an attribute, this would be more difficult.
Some more info on use case requirements for attributes would be useful.

**Comment by diegomasini** [comment]
@frankyclu as @smithbk mentioned the use of the yaml configuration file is temporary. I was talking with Keith about providing an API that allows a dynamic registration for attributes. The key point is to get proof about attribute ownership when values are being modified. It is worth mentioning that each attribute is encrypted with a different key and the same attribute in different TCerts will look different, to avoid linkability. If a key for a particular attribute is compromised that will allow the attacker to decrypt just that particular attribute, but not the rest in the same TCert nor in different TCerts. Attributes should be used encrypted, the functionality to do this is implemented but disabled for now. The ability to select which attributes to include in the TCert is implemented, the missing part is a public API that allows adding new or modifying existing attributes dynamically. Adding @andresgaragiola to this comments.

**Comment by andresgaragiola** [comment]
good question and good answers. I'm agreed with mentioned above. Just an additional comment. Suppose Alice has an account "0001" in the entity "Bank A" and other account "2003" in the entity "Bank B". If Alice is enrolled one time to Bank A and a second time to Bank B, then Alice has two ECerts and could generate TCerts to transact with "0001" or with "2003" independently. The good thing here is that chaincode developer don't has to worry about account.

**Comment by frankyclu** [comment]
I will work on this issue

providing API as @diegomasini described would add flexibility, but it will be an expensive process as it would require users making additional calls to PKI.

This issue does not necessary relates to attribute attestation (or using the affiliation capability) , as end users shall have the flexibility to pick the account IDs for themselves as long as they are unique and follows the formatting rules.

This is my work-around:

`
        attribute-entry-2: alice;bank_a;account1;22222-00001;2016-01-01T00:00:00-03:00;;
        attribute-entry-3: alice;bank_a;account2;22222-00002;2016-01-01T00:00:00-03:00;;
        attribute-entry-4: alice;bank_a;account3;22222-00003;2016-01-01T00:00:00-03:00;;
        attribute-entry-5: alice;bank_a;account4;22222-00004;2016-01-01T00:00:00-03:00;;
`
In short, user can specify attributes (Account ID) to use during TCert creation time

`aliceCert, err := alice.GetTCertificateHandlerNext("role", "account1", "account2", "account3")`

I have a sample example and will create a PR on that

I'll work on this issue
